### ISSUE TITLE
Allow to filter nested collection fields with *

### ISSUE THREAD

**Issue by franzwilding** [unite-cms/unite-cms#296]
Something like `SELECT * FROM content WHERE JSON_CONTAINS(data->"$.sections[*].title", JSON_QUOTE("My title"))` should work, however it returns null results. 

If we figure this out, a GraphQL filter like this: 

    filter: {
        field: "job_types[*].job_type.id",
        "operator: "=",
       "value": "..."
    }

should be possible.

**Comment by franzwilding** [comment]
One possible solution: https://dev.mysql.com/doc/refman/8.0/en/json-table-functions.html

**Comment by franzwilding** [comment]
@Clonk-HTML5

After thinking about different solutions, I might found something for you: You can run a LIKE comparison against a whole json_array in mysql. This way you can filter by collection children like this: 

```
query {
  findBirds(filter: {
    field: "features[*].feature",
    operator: "LIKE",
    value: "%\"spe%\"%"
  }) {
    result {
      features {
        feature
      }
    }
  }
}
```
This query will return all content nodes where the feature (text) field of the features (collection) field contains a string LIKE %spe%. 

I don't know if this is very performant, however if you run this against a small set of content it should be fine!

**Comment by Clonk-HTML5** [comment]
Ok this works for non referenced Content.
Just for info in Case you try to Filter for Reference an Issue exists #326.
Thanks

**Comment by franzwilding** [comment]
Filtering for referenced content is implemented in 0.10
